acceptance/lib/pxp-agent/bolt_pxp_module_helper.rb

```diff
@@ -108,29 +115,55 @@ def ensure_failed(broker, targets, response_dataset, max_retries: 30, query_inte
 # Runs a non-blocking task action on targets, and confirms that it was successful.
 # Block is executed on the stdout string.
 def run_successful_task(broker, targets, task, files, input: {}, metadata: nil, **kwargs, &block)
-  run_task(broker, targets, task, files, input, metadata, **kwargs) do |datas|
-    ensure_successful(broker, targets, datas, **kwargs, &block)
+  run_task(broker, targets, task, files, input: input, metadata: metadata) do |response_dataset|
+    ensure_successful(broker, targets, response_dataset, **kwargs, &block)
   end
 end
 
 # Runs a non-blocking task action on targets, and confirms that it failed but did not error.
 # Block is executed on the stderr string.
 def run_failed_task(broker, targets, task, files, input: {}, metadata: nil, **kwargs, &block)
-  run_task(broker, targets, task, files, input, metadata) do |datas|
-    ensure_failed(broker, targets, datas, **kwargs, &block)
+  run_task(broker, targets, task, files, input: input, metadata: metadata) do |response_dataset|
+    ensure_failed(broker, targets, response_dataset, **kwargs, &block)
   end
 end
 
 # Starts a blocking task on targets, and confirms that it produced an RPC error
 # Block is executed on the RPC error message string.
 def run_errored_task(broker, targets, task, files, input: {}, metadata: nil)
-  run_task(broker, targets, task, files, input, metadata,
+  run_task(broker, targets, task, files, input: input, metadata: metadata,
            blocking_request: true,
            expected_response_type: "http://puppetlabs.com/rpc_error_message") do |response_dataset|
     response_dataset.each { |data| yield data['description'] } if block_given?
   end
 end
 
+# Runs a non-blocking command action on targets, and confirms that it was successful.
+# Block is executed on the stdout string.
+def run_successful_command(broker, targets, command, **kwargs, &block)
+  run_command(broker, targets, command) do |response_dataset|
+    ensure_successful(broker, targets, response_dataset, **kwargs, &block)
+  end
+end
+
+# Runs a non-blocking command action on targets, and confirms that it unsuccessful, but did not error.
+# Block is executed on the stderr string.
+def run_failed_command(broker, targets, command, **kwargs, &block)
+  run_command(broker, targets, command) do |response_dataset|
+    ensure_failed(broker, targets, response_dataset, **kwargs, &block)
+  end
+end
+
+# Runs a non-blocking command action on targets, and confirms that it unsuccessful, but did not error.
```

Comment:
the comment says non-blocking, but the param on [L161](https://github.com/puppetlabs/pxp-agent/pull/754/commits/85058699d4a6346660e57bc9e40d755cf3262d98#diff-b29b1cee2f7efe4d0cf5fb1ab71bc054R161) says `blocking_request: true` ?

Reply:
Ah right, I fixed this in https://github.com/puppetlabs/pxp-agent/pull/761, but after I'd moved those changes from here to there. I've pushed a fix.